### DISCUSSION TITLE
Support for non-global filters

### DISCUSSION THREAD

**Discussion by sbauly** [alnitak/flutter_soloud#101] [General]
First of all, this is such a fantastic package - thanks so much to Marco and all of the contributors!

Currently, the only way to apply filters seems to be the `addGlobalFilter()` method. This is great for affecting all of the sounds at once, but doesn’t allow for more detailed control.

Are there plans to implement individual filters, or alternatively buses/sends? 

It would be very useful to be able to apply filters to certain sounds and not others, or have multiple sounds with varying degrees of wetness. For example, in a game, sounds that are further away might have more ‘wet’ reverb than closer sounds. Some sound fx may require an echo filter, whilst the background music is unchanged .etc. Thanks!

**Comment by alnitak** [comment]
Hi @sbauly, thanks again for your interest.

This is a good suggestion. I see it is possible to apply a filter to a single voice, the only drawback is that it will be active when setting it before playing it, but it is comprensible. I'll open a feat issue.

**Reply by sbauly** [reply]
Awesome, thank you 🙏

**Reply by te8uekm12** [reply]
This sounds fantastic! I’m really looking forward to this feature—any idea when it might be ready?

**Reply by alnitak** [reply]
Hey @te8uekm12, this is a feature already landed!

You can use a global filter using `SoLoud.instance.filter...` and sound filters like:
```
AudioSource source = SoLoud.instance.loadAsset('your/asset');
audioSource.filter...  // the filter must be set before playing it
SoLoud.instance.play(source);
```
